src/lro-poller.ts

```diff
@@ -0,0 +1,78 @@
+import { Queue } from "./throttler/queue";
+import * as api from "./api";
+import * as FirebaseError from "./error";
+
+export interface LroPollerOptions {
+  pollerName?: string;
+  apiOrigin: string;
+  apiVersion: "v1" | "v1beta1";
+  operationResourceName: string;
+  backoff?: number;
+  masterTimeout?: number;
+}
+
+export interface LroResult {
+  error?: FirebaseError;
+  response?: any;
+}
+
+const DEFAULT_INITIAL_BACKOFF_DELAY_MILLIS = 250;
+const DEFAULT_MASTER_TIMEOUT_MILLIS = 30000;
+
```

Comment:
Can we export a convenience method something like this:

```js
export function pollOperation(options: OperationPollerOptions, throttlerOptions?: ThrottlerOptions): Promise<OperationResult> {
  return new OperationPoller(throttlerOptions).poll(options);
}
```

Will make the common case more convenient.

That way you don't have to construct

Reply:
Done. but I would not let users of the method to decide the `throttlerOptions`

Reply:
LGTM. Not all `throttlerOptions` make sense in the OperationPoller, which is why we only expose a subset of them in `OperationPollerOptions`.

Can you switch the unit tests over to use this convenience method instead of directly using the OperationPoller's constructor?

Reply:
Done